AFQ/segmentation.py

```diff
@@ -125,165 +86,257 @@ def gaussian_weights(bundle, n_points=100, return_mahalnobis=False,
+        """
+        self.logger.info("Preparing Segmentation Parameters...")
+        self.prepare_img(fdata, fbval, fbvec, b0_threshold)
+        self.prepare_map(mapping, reg_prealign, reg_template)
+        self.bundle_dict = bundle_dict
+
+        self.logger.info("Preprocessing Streamlines...")
+        self.streamlines = streamlines
+        if self.nb_points > 0:
+            self.resample(self.nb_points)
+        if self.split:
+            self.split_streamlines()
+
+        return self.segment_roi(None, prob_threshold)
+
+    def prepare_img(self, fdata, fbval, fbvec, b0_threshold=0):
+        """
+        Prepare image data from DWI data.
+
+        Parameters
+        ----------
+        fdata, fbval, fbvec : str
+            Full path to data, bvals, bvecs
+        """
+        self.img, _, _, _ = \
+            ut.prepare_data(fdata, fbval, fbvec,
+                            b0_threshold=b0_threshold)
+        self.fdata = fdata
+        self.fbval = fbval
+        self.fbvec = fbvec
+
+    def prepare_map(self, mapping=None, reg_prealign=None, reg_template=None):
+        """
+        Set mapping between DWI space and a template.
+
+        Parameters
+        ----------
+        mapping : DiffeomorphicMap object, str or nib.Nifti1Image, optional.
+            A mapping between DWI space and a template.
+            If None, mapping will be registered from data used in prepare_img.
+            Default: None.
+
+        reg_template : str or nib.Nifti1Image, optional.
+            Template to use for registration (defaults to the MNI T2)
+        """
+        if reg_template is None:
+            reg_template = dpd.read_mni_template()
+
+        if mapping is None:
+            gtab = dpg.gradient_table(self.fbval, self.fbvec)
+            self.mapping = reg.syn_register_dwi(self.fdata, gtab)[1]
+        elif isinstance(mapping, str) or isinstance(mapping, nib.Nifti1Image):
+            if reg_prealign is None:
+                reg_prealign = np.eye(4)
+            self.mapping = reg.read_mapping(mapping, self.img, reg_template,
+                                            prealign=reg_prealign)
+        else:
+            self.mapping = mapping
+
+    def resample(self, nb_points, streamlines=None):
+        """
+        Resample streamlines to nb_points number of points.
+
+        Parameters
+        ----------
+        nb_points : int
+            Integer representing number of points wanted along the curve.
+            Streamlines will be resampled to this number of points.
+        streamlines : list of 2D arrays
+            Each array is a streamline, shape (3, N).
+            If streamlines is None, will use previously given streamlines.
+            Default: None.
+        """
+        if streamlines is None:
+            streamlines = self.streamlines
+
+        self.streamlines = np.array(
+            dps.set_number_of_points(streamlines, nb_points))
+
+    def split_streamlines(self, streamlines=None,
```

Comment:
We might need to rename this function, as it doesn't split streamlines (for now)